### PR TITLE
Generations Command "switch-generation" Becomes "checkout"

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This branch renames the `switch-generation` CLI command to `checkout`. It includes docs, variable and comments renaming.

The local store references to `ModelGeneration` are all updated to indicate `ActiveBranch`.

## QA steps

- Bootstrap.
- `export JUJU_DEV_FEATURE_FLAGS=generations`.
- `juju branch test-branch` -> Active branch set to "test-branch".
- `juju checkout master` -> Active branch set to "master".
- `juju checkout test-branch` -> Active branch set to "test-branch".
- `juju checkout not-there` -> ERROR this model has no active branch "not-there".

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Command "switch-generation" -> "checkout"](https://trello.com/c/8CPHoFTo/1261-command-switch-generation-checkout)

File: cmd/juju/model/checkout.go [hunk 1]
```diff
@@ -18,71 +18,77 @@ import (
 )
 
 const (
-	switchGenerationSummary = "Switch to the given generation."
-	switchGenerationDoc     = `
-Switch to the given generation, either current or next. 
+	checkoutSummary = "Work on the supplied branch."
+	checkoutDoc     = `
+Makes the supplied branch active, causing changes to charm configuration,
```
Comment (SimonRichardson) [inline cmd/juju/model/checkout.go]: I'm not sure I like the word active in this context, but I might be nit picking here, it implies that previously it was disabled.

Potential?
> Switch to the supplied branch

**Comment by manadart** [comment]
$$merge$$

File: cmd/juju/model/checkout.go (hunk 1 above)
Comment (manadart) [reply cmd/juju/model/checkout.go]: Changed it.

**Pull request merged**